src/utils/validation-rules.js

```diff
@@ -31,6 +29,32 @@ export function useTextAreaFieldRules() {
   ];
 }
 
+export function useBarCodeValidationRules(value) {
+  const regex = /^[0-9a-zA-Z!@#$%^&*()_'"+-={}[\]:;<>,.?~\\s]+$/;
+
+  const rules = [
+    { message: "Input field is required", isValid: !!value },
+    {
+      message: "Only English letters and signs are allowed",
+      isValid: regex.test(value),
+    },
+    {
+      message: "Input must be less than 100 characters",
+      isValid: value && value.length <= 100,
+    },
+  ];
+
+  const isValid = rules.every((rule) => rule.isValid);
+
+  return {
+    messages: rules
+      .map((rule) => (!rule.isValid ? rule.message : null))
+      .filter((msg) => msg)
+      .join(", "),
+    isValid,
+  };
+}
```

Comment:
A potential area for improvement is the performance aspect of creating the messages string. Currently, it iterates over all rules twice: once for checking validity (every) and once for constructing the message string (map followed by filter). This is not a significant concern for a small number of rules but could be optimized for readability and efficiency like this:
```
export function useBarCodeValidationRules(value) {
  const regex = /^[0-9a-zA-Z!@#$%^&*()_'"+-={}[\]:;<>,.?~\\s]+$/;
  let messages = [];
  let isValid = true;

  const addRule = (message, isValidCondition) => {
    if (!isValidCondition) {
      messages.push(message);
      isValid = false;
    }
  };

  addRule("Input field is required", !!value);
  addRule("Only English letters and signs are allowed", regex.test(value));
  addRule("Input must be less than 100 characters", value && value.length <= 100);

  return {
    messages: messages.join(", "),
    isValid,
  };
}
```